### ISSUE TITLE
Are Keybase.io generated GPG keys vulnerable to the ROCA CVE?

### ISSUE THREAD

**Issue by dragon788** [keybase/client#9162]
https://crocs-muni.github.io/roca/

TL;DR Keys generated by certain (Infineon) hardware devices are easily reversible to generate a private key from only public key material. If an OS or browser uses any hardware assisted encryption it could produce a vulnerable key (this is conjecture). This only applies to keys <4096 bits, though they assume if generated on the same hardware those may also be vulnerable, they just require extremely expensive amounts of CPU and more time than is feasible to test with current hardware.

I don't know if you guys use an HSM or anything to accelerate key generation or if they are generated client side, but it might be good to scan the public keys on Keybase and alert potentially vulnerable users. See the article on services available to test whether keys are in fact vulnerable.

**Comment by maxtaco** [comment]
We'll have an update on this soon, but for now, definitely not, no keys we generated are vulnerable to ROCA. On the website, we use [browser-side JS](https://github.com/keybase/kbpgp) to generate RSA keys, and on the client we just use the [standard Go implementation](https://golang.org/pkg/crypto/rsa/), which is in software.

**Comment by dragon788** [comment]
Good to hear you guys are on top of it. Hopefully it is possible to warn susceptible users and help them regen to avoid impersonation attacks, and I love how your "chain of identity" is awesome at proving when the "real" person requested the change.
